Charon/filetypes/UltimakerFormatPackage.py

```diff
@@ -418,9 +418,10 @@ def _readMetadata(self) -> None:
                 self._readMetadataElement(metadata, metadata_file)
 
         if self.mode != OpenMode.WriteOnly and not self.getMetadata("/3D/model.gcode"):
-            gcode_stream = TextIOWrapper(self.zipfile.open("/3D/model.gcode"))
-            header_data = GCodeFile.parseHeader(gcode_stream, prefix = "/3D/model.gcode/")
-            self.metadata.update(header_data)
+            if "/3D/model.gcode" in [member.filename for member in self.zipfile.infolist()]:
```

Comment:
Since we are just checking "does an entry named /3D/model.gcode exist", wouldn't it be better to use something like:

```
try:
  self.zipfile.getinfo("/3D/model.gcode")
except KeyError:
  return
```

Reply:
Fixed.